packages/darklang/languageTools/writtenTypesToProgramTypes.dark

```diff
@@ -344,6 +344,19 @@ module Darklang =
                 let name = name |> Stdlib.List.last |> Builtin.unwrap
                 ProgramTypes.PipeExpr.EPipeVariable(gid (), name, typeArgs)
 
+        let stringSegmentToPT
+          (onMissing: NameResolver.OnMissing)
+          (pm: ProgramTypes.PackageManager.PackageManager)
+          (owner: String)
+          (currentModule: List<String>)
+          (segment: WrittenTypes.StringSegment)
+          : ProgramTypes.StringSegment =
+          match segment with
+          | StringText(_, s) -> ProgramTypes.StringSegment.StringText s
+          | StringInterpolation(_, expr, _, _) ->
+            ProgramTypes.StringSegment.StringInterpolation(
+              Expr.toPT onMissing pm owner currentModule expr
+            )
```

Comment:
might be cleaner to inline this in the `Expr.toPT`
- LetPattern is simple, but used by both ELambda and ELet, so extracted is good
- MatchPattern and PipeExpr are only used in one place each (I think) but are more complicated so good to extract